model/secret.go

```diff
@@ -20,5 +20,10 @@ package model
 // TODO: add controller hooks for notification about changes to the secret
 type SecretRegistry interface {
 	// GetSecret retrieves the secret data by a platform specific URI
-	GetSecret(uri string) (map[string][]byte, error)
+	GetSecret(uri string) (*TLSContext, error)
```

Comment:
It would also be good to rename this method to `GetCertificate(host string)`. This clearly captures what this method does and on what parameters it operates on, compared to `GetSecret(uri)`.